python/hail/table.py

```diff
@@ -1389,22 +1377,22 @@ def collect(self):
         :obj:`list` of :class:`.Struct`
             List of rows.
         """
-        return hl.tarray(self.schema)._from_json(self._jt.collectJSON())
+        return hl.tarray(self.row.dtype)._from_json(self._jt.collectJSON())
 
     def describe(self):
         """Print information about the fields in the table."""
 
         def format_type(typ):
             return typ.pretty(indent=4)
 
-        if len(self.global_schema) == 0:
+        if len(self.globals.dtype) == 0:
             global_fields = '\n    None'
         else:
             global_fields = ''.join("\n    '{name}': {type} ".format(
-                name=f, type=format_type(t)) for f, t in self.global_schema.items())
+                name=f, type=format_type(t)) for f, t in self.globals.dtype.items())
 
         row_fields = ''.join("\n    '{name}': {type} ".format(
-            name=f, type=format_type(t)) for f, t in self.schema.items())
+            name=f, type=format_type(t)) for f, t in self.row.dtype.items())
```

Comment:
This should be symmetric with other dimensions in printing `None` if len(row_fields) == 0.